src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1608,43 +1614,48 @@ else if (toUpdate.getContentAccessMode() != null) {
+        List<EnvironmentDTO> environments = updated.getEnvironments();
+
+        if (environments == null) {
             return false;
         }
-
         boolean changesMade = false;
-        String environmentId = updated.getEnvironment() == null ? null : updated.getEnvironment().getId();
-        String environmentName = updated.getEnvironment() == null ? null : updated.getEnvironment().getName();
-        String validatedEnvId = null;
+        Map<String, String> validatedEnvIds = new HashMap<>();
 
-        if (environmentId != null) {
-            if (toUpdate.getEnvironmentId() == null || !toUpdate.getEnvironmentId().equals(environmentId)) {
+        if (environments != null) {
+            for (EnvironmentDTO environment : environments) {
+                String environmentId = environment == null ? null : environment.getId();
+                String environmentName = environment == null ? null : environment.getName();
 
-                if (!environmentCurator.exists(environmentId)) {
-                    throw new NotFoundException(i18n.tr(
-                        "Environment with ID \"{0}\" could not be found.", environmentId));
+                if (environmentId != null) {
+                    if (!environmentCurator.exists(environmentId)) {
+                        throw new NotFoundException(i18n.tr(
+                            "Environment with ID \"{0}\" could not be found.", environmentId));
+                    }
                 }
+                else if (environmentName != null) {
+                    environmentId = this.environmentCurator.
+                        getEnvironmentIdByName(toUpdate.getOwnerId(), environmentName);
 
-                validatedEnvId = environmentId;
-            }
-
-        }
-        else if (environmentName != null) {
-            String envId = this.environmentCurator.
-                getEnvironmentIdByName(toUpdate.getOwnerId(), environmentName);
+                    if (environmentId == null) {
+                        throw new NotFoundException(i18n.tr(
+                            "Environment with name \"{0}\" could not be found.", environmentName));
+                    }
+                }
+                else {
+                    throw new NotFoundException(i18n.tr("Environment could not be found."));
+                }
 
-            if (envId == null) {
-                throw new NotFoundException(i18n.tr(
-                    "Environment with name \"{0}\" could not be found.", environmentName));
-            }
+                if (validatedEnvIds.containsKey(environmentId)) {
+                    throw new ConflictException("Environment \"{0}\" already exists: " + environmentId);
+                }
 
-            if (toUpdate.getEnvironmentId() == null || !envId.equals(toUpdate.getEnvironmentId())) {
-                validatedEnvId = envId;
+                validatedEnvIds.put("" + validatedEnvIds.size(), environmentId);
```

Comment:
Don't use string concatenation for this type of logic; things like `String.valueOf` or `.toString` would be better options.

Reply:
Done.